blocks/components/editable/index.js

```diff
@@ -306,14 +305,35 @@ export default class Editable extends wp.element.Component {
 		return !! this.state.formats[ format ];
 	}
 
-	toggleFormat( format ) {
-		this.editor.focus();
-
-		if ( this.isFormatActive( format ) ) {
-			this.editor.formatter.remove( format );
-		} else {
-			this.editor.formatter.apply( format );
+	changeFormats( formats ) {
+		if ( this.state.bookmark ) {
+			this.editor.selection.moveToBookmark( this.state.bookmark );
 		}
+
+		forEach( formats, ( formatValue, format ) => {
+			if ( format === 'link' ) {
+				if ( formatValue !== undefined ) {
+					const anchor = this.editor.dom.getParent( this.editor.selection.getNode(), 'a' );
+					if ( ! anchor ) {
+						this.editor.formatter.remove( 'link' );
+					}
+					this.editor.formatter.apply( 'link', { href: formatValue.value }, anchor );
```

Comment:
Should this occur if `! anchor` condition above passes and the link is removed? Seems like this ought to be:

```js
if ( anchor ) {
	// apply link
} else {
	// remove link
}
```